sessions-core/src/main/java/com/adyen/checkout/sessions/repository/SessionRepository.kt

```diff
@@ -43,135 +40,93 @@ internal class SessionRepository(
         configuration: Configuration,
         session: Session,
         order: OrderRequest?
-    ): SessionSetupResponse {
+    ): Result<SessionSetupResponse> = runCatching {
```

Comment:
A small note about `runCatching`, it might swallow `CancellationException`
https://github.com/Kotlin/kotlinx.coroutines/issues/1814

Reply:
Ohh didn't know that. Should we create a `runSuspendCatching` like some suggest in that thread?

Reply:
Yeah I think that would be best.